### PR TITLE
#170947559 Avail requests to approve

### PR DESCRIPTION
**Pull request by Ntare22**
#### What does this PR do?
- Avails requests for a manager to approve or reject
#### Description of Task to be completed?
- adjusted trip request to include managerId
- created middleware function to check if user checking is a manager
- created controller to availTripRequests

#### How should this be manually tested?
- Login as a manager, `umuhozad@andela.com`, password, `Bien@BAR789`
- Put the token provided in the headers.
- In postman, using a GET request, enter this link `localhost:8000/api/v1/trips/requests/view`
- Click send

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
[#170947559](https://www.pivotaltracker.com/story/show/170947559)

#### Screenshots (if appropriate)
image

#### Questions:
- N/A

### CODE REVIEW

File: src/routes/tripsRoutes.js [hunk 1]
```diff
@@ -6,8 +6,9 @@ import protectRoute from '../middlewares/protectRoute';
 
 const router = express.Router();
 
-router.post('/one-way', protectRoute.verifyUser, protectRoute.verifyRequester, requestRules, validationResult, tripsController.createRequest);
-router.post('/return', protectRoute.verifyUser, protectRoute.verifyRequester, returnTripRules, validationResult, tripsController.createReturnRequest);
-router.patch('/edit', protectRoute.verifyUser, protectRoute.verifyRequester, requestRules, validationResult, tripsController.editRequest);
+router.post('/one-way', protectRoute.verifyUser, protectRoute.verifyRequester, protectRoute.checkUserManager, requestRules, validationResult, tripsController.createRequest);
+router.post('/return', protectRoute.verifyUser, protectRoute.verifyRequester, protectRoute.checkUserManager, returnTripRules, validationResult, tripsController.createReturnRequest);
+router.patch('/edit', protectRoute.verifyUser, protectRoute.verifyRequester, protectRoute.checkUserManager, requestRules, validationResult, tripsController.editRequest);
+router.get('/requests/view', protectRoute.verifyUser, protectRoute.verifyManager, tripsController.availTripRequests);
```
Comment (Mnickii) [inline src/routes/tripsRoutes.js]: how about just ```/view```

**Review by Mnickii — CHANGES_REQUESTED** [review]

**Review by Mnickii — DISMISSED** [review]
LGTM

**Review by Baraka-Mugisha — APPROVED** [review]

File: src/swagger/trips.swagger.js [hunk 2]
```diff
@@ -294,3 +294,27 @@
 *       '400':
 *             description: Please double checks your departure and arrival dates.
 * */
+
+/**
+ * @swagger
+ * /api/v1/trips/requests/view:
```
Comment (Baraka-Mugisha) [inline src/swagger/trips.swagger.js]: We could just remove `requests/` since we are not using it.

**Review by Baraka-Mugisha — CHANGES_REQUESTED** [review]

File: src/swagger/trips.swagger.js (hunk 2 above)
Comment (Ntare22) [reply src/swagger/trips.swagger.js]: I've worked on it.

File: src/controllers/tripsController.js [hunk 3]
```diff
@@ -52,6 +54,7 @@ export default class requestController {
         return Response.success(res, 201, 'Request created successfully', newRequest);
       }
     } catch (error) {
+      // console.log(error);
```
Comment (izzett222) [inline src/controllers/tripsController.js:57]: could you remove those commented code

**Review by izzett222 — CHANGES_REQUESTED** [review]

File: src/controllers/tripsController.js (hunk 3 above)
Comment (Ntare22) [reply src/controllers/tripsController.js:57]: I've worked on it.

**Review by izzett222 — APPROVED** [review]

**Review by Mnickii — APPROVED** [review]

**Review by Baraka-Mugisha — APPROVED** [review]

**Pull request merged**